data-prepper-plugins/parse-json-processor/src/main/java/org/opensearch/dataprepper/plugins/codec/json/JsonInputCodec.java

```diff
@@ -3,29 +3,28 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-package org.opensearch.dataprepper.plugins.source.codec;
+package org.opensearch.dataprepper.plugins.codec.json;
 
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.codec.InputCodec;
 import org.opensearch.dataprepper.model.event.Event;
 import org.opensearch.dataprepper.model.event.JacksonEvent;
 import org.opensearch.dataprepper.model.log.JacksonLog;
 import org.opensearch.dataprepper.model.record.Record;
-import com.fasterxml.jackson.core.JsonFactory;
-import com.fasterxml.jackson.core.JsonParser;
-import com.fasterxml.jackson.core.JsonToken;
-import com.fasterxml.jackson.databind.ObjectMapper;
 
 import java.io.IOException;
 import java.io.InputStream;
 import java.util.Map;
 import java.util.Objects;
 import java.util.function.Consumer;
 
-/**
- * An implementation of {@link Codec} which parses JSON objects for arrays.
- */
-@DataPrepperPlugin(name = "json", pluginType = Codec.class)
-public class JsonCodec implements Codec {
+@DataPrepperPlugin(name = "json", pluginType = InputCodec.class)
```

Comment:
Structure wise instead of having InputCodecs spread around (i.e. some are under `data-prepper-plugins`, this one is under the json processor, etc), can we have a single plugins directory for all InputCodecs. Something like

```
data-prepper-plugins/
      input-codecs/
          csv-codec/
          parquet-codec/
          json-codec/
          newline-codec/
          avro-codec/
```

Reply:
Repository structure was discussed with David at length. The reason that those codecs are placed in those projects (for example csv codec in csv parse processor and so on) is so that they can share some common dependencies.

Reply:
Yes, I think if we combine these we can reduce code duplication in how we handle the CSV/JSON.

Original comment on it:

https://github.com/opensearch-project/data-prepper/pull/2309#discussion_r1121930990